src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -854,6 +852,9 @@ protected int engineDoFinal(ByteBuffer input, ByteBuffer output) throws ShortBuf
     }
 
     private void checkOutputBuffer(int inputLength, byte[] output, int outputOffset) throws ShortBufferException {
+        if (inputLength < 0 || outputOffset < 0 || outputOffset > output.length) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
```

Comment:
Given that we added this condition, it would be good to have a test for this condition.

Reply:
Tested by existing tests. That's how I discovered this new line was needed.